From 79525de509ccb107e388c5f4d5bb700b2d1e8831 Mon Sep 17 00:00:00 2001
From: Shiz <hi@shiz.me>
Date: Sat, 20 May 2017 18:32:45 +0200
Subject: [PATCH] abuild: add !checkroot option to run tests without fakeroot

Due to fakeroot being very permissive with regards to file permissions,
some package testsuites that explicitly check for failing permission checks
will fail.

For those testsuites, give the APKBUILD the option to run the tests outside
of a fakeroot environment.
---
 abuild.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/abuild.in b/abuild.in
index c003282..0f54c87 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1401,6 +1401,9 @@ build_abuildrepo() {
 		# if package() is missing then build is called from rootpkg
 		_build=true
 	fi
+	if options_has "!checkroot"; then
+		_check=check
+	fi
 	if ! want_check; then
 		_check=true
 	fi
